federation/src/graphql/api/1_0/resolver/AuthenticationResolver.ts

```diff
@@ -1,19 +1,31 @@
-import { CONFIG } from '@/config'
 import { LOG4JS_BASE_CATEGORY_NAME } from '@/config/const'
-import { EncryptedTransferArgs, interpretEncryptedTransferArgs } from 'core'
+import { CommunityHandshakeStateLogic, EncryptedTransferArgs, interpretEncryptedTransferArgs, splitUrlInEndPointAndApiVersion } from 'core'
 import {
-  CommunityLoggingView,
-  Community as DbCommunity,
+  CommunityHandshakeStateLoggingView,
+  CommunityHandshakeState as DbCommunityHandshakeState,
+  CommunityHandshakeStateType,
   FederatedCommunity as DbFedCommunity,
-  FederatedCommunityLoggingView,
   getHomeCommunity,
+  findPendingCommunityHandshakeOrFailByOneTimeCode,
+  getCommunityByPublicKeyOrFail,
 } from 'database'
 import { getLogger } from 'log4js'
-import { AuthenticationJwtPayloadType, AuthenticationResponseJwtPayloadType, encryptAndSign, OpenConnectionCallbackJwtPayloadType, OpenConnectionJwtPayloadType, uint32Schema, uuidv4Schema } from 'shared'
+import { 
+  AuthenticationJwtPayloadType, 
+  AuthenticationResponseJwtPayloadType, 
+  Ed25519PublicKey, 
+  encryptAndSign, 
+  OpenConnectionCallbackJwtPayloadType, 
+  OpenConnectionJwtPayloadType, 
+  uint32Schema, 
+  uuidv4Schema 
+} from 'shared'
 import { Arg, Mutation, Resolver } from 'type-graphql'
 import { startAuthentication, startOpenConnectionCallback } from '../util/authenticateCommunity'
 
-const createLogger = (method: string  ) => getLogger(`${LOG4JS_BASE_CATEGORY_NAME}.graphql.api.1_0.resolver.AuthenticationResolver.${method}`)
+// TODO: think about the case, when we have a higher api version, which still use this resolver
+const apiVersion = '1_0'
```

Comment:
In the core-module config the config-key FEDERATION_BACKEND_SEND_ON_API exists to define on which api-version the instance will work.
If we use this config-key at any place of usage of api-version it should solve the question yet.